server/sockets/socket.js

```diff
@@ -0,0 +1,48 @@
+/* All of the machinery required to work our websockets. */
+const ws = require("ws");
+const url = require("url");
+const querystring = require("querystring");
+const socketActions = require("./socketActions");
+
+const socketServers = {};
+
+function makeSocket(name) {
+  const socket = new ws.Server({ noServer: true });
+  socket.on("connection", (ws, request, client) => {
+    ws.on("message", (msg) => {
+      const message = JSON.parse(msg);
+      if (message.type && message.type in socketActions) {
```

Comment:
Extra `message.type` here?

Reply:
Nah, the first check is to make sure it's not null and the second is to see if it's a key in the dictionary.

If the dictionary somehow has a null value as a key and the `message.type` is also null, without the check the subsequent function call off of the dictionary could fail and crash the server

Reply:
Oh I completely brain farted on this lol